File: application/backend-credit-card-enrollment/backend-java/src/main/java/cloud/ambar/common/queryhandler/QueryHandler.java
package cloud.ambar.common.queryhandler;

import cloud.ambar.common.projection.MongoTransactionalProjectionOperator;
import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
abstract public class QueryHandler {
    final protected MongoTransactionalProjectionOperator mongoTransactionalProjectionOperator;
    public abstract Object handleQuery(Query query);
}

File: application/backend-credit-card-enrollment/backend-c#/CreditCardEnrollment/Common/Query/QueryHandler.cs
using CreditCardEnrollment.Common.Projection;

namespace CreditCardEnrollment.Common.Query;

public interface IQueryHandler<in TQuery, TResult> where TQuery : IQuery<TResult>
{
    Task<TResult> Handle(TQuery query);
}

public abstract class QueryHandler<TQuery, TResult>(IMongoTransactionalProjectionOperator mongoOperator)
    : IQueryHandler<TQuery, TResult>
    where TQuery : IQuery<TResult>
{
    protected readonly IMongoTransactionalProjectionOperator MongoOperator = mongoOperator;

    public abstract Task<TResult> Handle(TQuery query);
}
